plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiUtil.java

```diff
@@ -178,6 +168,30 @@ public static Schema constructSchema(List<String> columnNames, List<HiveType> co
         return fieldBuilder.endRecord();
     }
 
+    public static Schema constructSchema(Schema dataSchema, List<String> columnNames)
```

Comment:
Is this only for handling the data schema for reading log files?

Reply:
Not exactly, parquet will also use this schema. The `HudiTrinoReaderContext` will invoke serialize too, which will be building a GenericRecord with the data schema that is returned by `#constructSchema`.

I didn't really dig deeply into why there's no error when handling values from parquet. But judging from the error stack, it seems that this schema is treated as the `writerSchema` and used to read the log, which is why the error is thrown in the presence of log files.